test/test_nn.py

```diff
@@ -11494,6 +11494,50 @@ def __init__(self):
                 self.assertEqual(p.grad.to(devices[0]), pe.grad)
 
 
+class TestLazyModules(TestCase):
+    def test_lazy_module(self):
+        module = torch.nn.Module()
+        module.register_parameter('test_param', torch.nn.parameter._UninitializedParameter())
+
+        with self.assertRaisesRegex(ValueError, 'uninitialized'):
+            list(module.parameters())
+
+        with self.assertRaisesRegex(ValueError, 'uninitialized'):
+            module._apply(lambda x: x)
+
+        state_dict = module.state_dict()
+        self.assertIsInstance(state_dict['test_param'], nn.parameter._UninitializedParameter)
+        new_module = torch.nn.Module()
+        # Uninitialized paramaters in a state dict are ignored when the module has a valid one.
```

Comment:
What about the other way around when `new_module` has undefined and the state dict has a full weight?

Reply:
I was lazy to write it in the first pass, will add it